src/DssSpell.sol

```diff
@@ -16,44 +16,57 @@
 pragma solidity 0.5.12;
 
 import "lib/dss-interfaces/src/dapp/DSPauseAbstract.sol";
-import "lib/dss-interfaces/src/dss/VatAbstract.sol";
+import "lib/dss-interfaces/src/dapp/DSAuthorityAbstract.sol";
+import "lib/dss-interfaces/src/dss/OsmMomAbstract.sol";
+import "lib/dss-interfaces/src/dss/FlipperMomAbstract.sol";
 import "lib/dss-interfaces/src/dss/ChainlogAbstract.sol";
 
 contract SpellAction {
     // MAINNET ADDRESSES
     //
     // The contracts in this list should correspond to MCD core contracts, verify
     //  against the current release list at:
-    //     https://changelog.makerdao.com/releases/mainnet/1.1.5/contracts.json
+    //     https://changelog.makerdao.com/releases/mainnet/active/contracts.json
     ChainlogAbstract constant CHANGELOG = ChainlogAbstract(0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F);
 
-    // Decimals & precision
-    uint256 constant THOUSAND = 10 ** 3;
-    uint256 constant MILLION  = 10 ** 6;
-    uint256 constant WAD      = 10 ** 18;
-    uint256 constant RAY      = 10 ** 27;
-    uint256 constant RAD      = 10 ** 45;
-
-    // Many of the settings that change weekly rely on the rate accumulator
-    // described at https://docs.makerdao.com/smart-contract-modules/rates-module
-    // To check this yourself, use the following rate calculation (example 8%):
-    //
-    // $ bc -l <<< 'scale=27; e( l(1.08)/(60 * 60 * 24 * 365) )'
-    //
-    // A table of rates can be found at
-    //    https://ipfs.io/ipfs/QmefQMseb3AiTapiAKKexdKHig8wroKuZbmLtPLv4u2YwW
+    address constant MCD_ADM            = 0x0a3f6849f78076aefaDf113F5BED87720274dDC0;
+    address constant VOTE_PROXY_FACTORY = 0x6FCD258af181B3221073A96dD90D1f7AE7eEc408;
 
     function execute() external {
-        // Proving the Pause Proxy has access to the MCD core system at the execution time
-        address MCD_VAT = CHANGELOG.getAddress("MCD_VAT");
-        require(VatAbstract(MCD_VAT).wards(address(this)) == 1, "no-access");
+        address MCD_PAUSE   = CHANGELOG.getAddress("MCD_PAUSE");
+        address FLIPPER_MOM = CHANGELOG.getAddress("FLIPPER_MOM");
+        address OSM_MOM     = CHANGELOG.getAddress("OSM_MOM");
+
+        // Change MCD_ADM address in the changelog (Chief)
+        CHANGELOG.setAddress("MCD_ADM", MCD_ADM);
+
+        // Add VOTE_PROXY_FACTORY to the changelog (previous one was missing)
+        CHANGELOG.setAddress("VOTE_PROXY_FACTORY", VOTE_PROXY_FACTORY);
+
+        // Bump version
+        CHANGELOG.setVersion("1.2.0");
+
+        // Set new Chief in the Pause
+        DSPauseAbstract(MCD_PAUSE).setAuthority(MCD_ADM);
+
+        // Set new Chief in the FlipperMom
+        FlipperMomAbstract(FLIPPER_MOM).setAuthority(MCD_ADM);
+
+        // Set new Chief in the OsmMom
+        OsmMomAbstract(OSM_MOM).setAuthority(MCD_ADM);
+
+        // Set Pause delay to 12 hours
```

Comment:
Personally, I think 12 hours is a little short for reacting to a governance attack. Further, I suspect that for most other problems (e.g. a market crash), there's no material difference between 12 and 24 hours; problems like that are why we have Moms, which are being re-enabled with this update.

IMO 24 hours is the sweet spot, and leaving it at 3 days is the second-best option. The fact that there was no polling or signal requests here is something to flag--making a change to the delay that wasn't discussed in enough detail could jeopardize the passing of this executive.

Reply:
Exactly, I'd suggest keeping it 3 days if we are not setting it back to 12 hours.

Reply:
Then let's just stick with three days and the usual discussion/signal request/polling procedure can be followed to select the next value after the migration completes.

Reply:
Sounds good, will get the delay change removed.